src/type.js

```diff
@@ -75,70 +152,11 @@ function getStateType(value) {
 }
 
 /**
- * Creates a validator that compares a value to a specific class.
- * @param {!function} expectedClass Class to check value against.
- * @return {function} Validator.
- */
-function instanceOfValidatorFactory(expectedClass) {
-	return (value, name, context) => {
-		if (!(value instanceof expectedClass)) {
-			return composeError(`Expected instance of ${expectedClass}`, name, context);
-		}
-
-		return true;
-	};
-}
-
-/**
- * Creates a validator that checks the values of an object against a type.
- * @param {!function} typeValidator Validator to check value against.
- * @return {function} Validator.
- */
-function objectOfValidatorFactory(typeValidator) {
-	return (value, name, context) => {
-		let success = true;
-
-		for (let key in value) {
-			success = !(typeValidator(value[key], null) instanceof Error);
-		}
-
-		if (!success) {
-			return composeError('Expected object of one type', name, context);
-		}
-
-		return true;
-	};
-}
-
-/**
- * Creates a validator that checks a value against multiple types and only has to pass one.
- * @param {!array} arrayOfTypeValidators Array of validators to check value against.
- * @return {function} Validator.
- */
-function oneOfTypeValidatorFactory(arrayOfTypeValidators) {
-	if (!Array.isArray(arrayOfTypeValidators)) {
-		return (value, name, context) => composeError('Expected an array.', name, context);
-	}
-
-	return (value, name, context) => {
-		for (let i = 0; i < arrayOfTypeValidators.length; i++) {
-			const validator = arrayOfTypeValidators[i];
-
-			if (!(validator(value, name) instanceof Error)) {
-				return true;
-			}
-		}
-
-		return composeError('Expected one of given types.', name, context);
-	};
-}
-
-/**
  * Creates a validator that checks against a specific primitive type.
  * @param {!string} expectedType Type to check against.
```

Comment:
Another thing in google's standard is that primitive types are [non-nullable by default](https://developers.google.com/closure/compiler/docs/js-for-compiler#nonnull), so they don't need to use the `!` operator (though it's not a problem using it).